src/controller/model/endpoint.ts

```diff
@@ -310,6 +310,48 @@ class Endpoint extends Entity {
         }
     }
 
+    public async writeUndiv(
+        clusterKey: number | string, attributes: KeyValue, options?: Options
+    ): Promise<void> {
+        options = this.getOptionsWithDefaults(options, true, Zcl.Direction.CLIENT_TO_SERVER);
+        const cluster = Zcl.Utils.getCluster(clusterKey);
+        const payload: {attrId: number; dataType: number; attrData: number| string | boolean}[] = [];
+        for (const [nameOrID, value] of Object.entries(attributes)) {
+            if (cluster.hasAttribute(nameOrID)) {
+                const attribute = cluster.getAttribute(nameOrID);
+                payload.push({attrId: attribute.ID, attrData: value, dataType: attribute.type});
+            } else if (!isNaN(Number(nameOrID))){
+                payload.push({attrId: Number(nameOrID), attrData: value.value, dataType: value.type});
+            } else {
+                throw new Error(`Unknown attribute '${nameOrID}', specify either an existing attribute or a number`);
+            }
+        }
+
+        const log = `WriteUndiv ${this.deviceIeeeAddress}/${this.ID} ` +
+            `${cluster.name}(${JSON.stringify(attributes)}, ${JSON.stringify(options)})`;
+        debug.info(log);
+
+        try {
+            const frame = Zcl.ZclFrame.create(
+                Zcl.FrameType.GLOBAL, options.direction, options.disableDefaultResponse,
+                options.manufacturerCode, options.transactionSequenceNumber ?? ZclTransactionSequenceNumber.next(),
+                'writeUndiv', cluster.ID, payload, options.reservedBits
```

Comment:
Seems this is the only diff compared to the `write()`. Please add an option to `options` called `writeUndiv`, if true writeUndiv will be used. This avoid a lot of code duplication. Also write a test for this in `controller.test.ts` since new code has to be covered by tests (otherwise build fails).

Reply:
I'm afraid that livolo uses not only "write" (Id: 02) and "writeUndiv" (Id: 03) command types. 
In sniff log I found also "readRsp" (Id: 01) command.

Is it ok to add and option commandKey: number | string with default value "write", that would accept any value declared in foundation.ts?

Reply:
About the `readRsp`, can't this be done with: https://github.com/Koenkk/zigbee-herdsman/blob/master/src/controller/model/endpoint.ts#L352 ?

Reply:
At glance readResponse() can be reused, not tested yet as those packets are still need to be researched.
Do I need to rework commandKey option (number|string) to writeUndiv (boolean) or current implementation (see updated pull request) is ok?

Reply:
Please refactor to boolean (since other commandKey make no sense)

Reply:
Done, reworked and added extra coverage test for writeUndiv.